### PR TITLE
Propose code owners from Intel

### PR DESCRIPTION
**Pull request by jgong5**
This PR proposes a list of CPU-related PyTorch modules that Intel is willing to own or co-own.

### CODE REVIEW

**Comment by jgong5** [comment]
The proposed list contains modules that fall into the following categories:
1. Intel was the primary contributor and is willing to own the modules (“CPU frontend”, “oneDNN graph integration” and “IDEEP”).
2. Intel made significant contributions and is willing to co-own with Meta and other parties (“CPU ATen backend”, “Autocast”).
3. Intel just started contributing, plans to put major effort and becomes the major contributor. Intel is willing to co-own with Meta (“NNC”).

**Comment by jgong5** [comment]
@malfet I applied the changes in CODEOWNERS to the merge_rules.json. Could you please review? For modules in category 2 and 3, there should be Meta owners and/or third-party owners too but I am not sure the appropriate persons to add. Please suggest anything we need to revise. Also, do we need to revert the changes in CODEOWNERS?

**Review by malfet — CHANGES_REQUESTED** [review]
Let's separate this one from CODEOWNER and exclude NCC for now (I remember @VitalyFedyunin had some concerns about it)

Also, I'm not sure if submodule update would work as `ideep/**` pattern, but we can give it a try

File: .github/merge_rules.json [hunk 1]
```diff
@@ -188,6 +188,90 @@
          "Lint",
          "pull"
       ]
+    },
+    {
+      "name": "IDEEP",
+      "patterns": [
+        "third-party/ideep/**",
```
Comment (malfet) [inline .github/merge_rules.json:195]: I though it should be something like the following
```suggestion
        "third_party/ideep",
```

**Comment by jgong5** [comment]
@malfet I removed NNC from the merge_rules.json and reverted changes from CODEOWNERS. Please review. Thanks.

**Comment by jgong5** [comment]
Hi @malfet Does the PR look good to you? Any more comments?

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**